handler/EventHandler.go

```diff
@@ -0,0 +1,209 @@
+		Time:          timeNow,
+		Message:       e.EventMsg,
+		MessageOffset: "0",
+	})
+
+	key := []byte(e.InstanceID)
+	value := []byte(e.EventMsg)
+	expire := 25
+
+	check := arraySlicing(instanceEvents, *events)
+	if check {
+		_, err := c.Get(key)
+		if err != nil {
+			c.Set(key, value, expire)
+		} else {
+			c.Del(key)
+		}
+	}
+
+	got, err := c.Get(key)
+	if err != nil {
+		c.Set(key, value, expire)
+		err := e.putEvent(httpClient, jsonUpload, cfg, status)
+		if err != nil {
+			return err
+		} else {
+			return nil
+		}
+	} else {
+		log.Info("This instance is saved inmemory already.", log.Data{e.InstanceID: "this instance is saved inmemory", "lock": got})
+		return nil
+	}
+}
+
+/*this puts an event into the database under the instance you chose
+it does some checks to make sure the instance exists and checks the status
+if the status isn't already failed it will turn that instance to failed */
+func (e *EventReport) putEvent(httpClient *http.Client, json []byte, cfg *config.Config, status string) error {
+
+	path := cfg.ImportAPIURL + "/instances/" + e.InstanceID + "/events"
+
+	var URL *url.URL
+	URL, err := url.Parse(path)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest("PUT", URL.String(), bytes.NewBuffer(json))
+	if err != nil {
+		return err
+	}
+	log.Info(cfg.ImportAuthToken, nil)
+	//this needs to change but for testing purposes leave it in
+	req.Header.Set("Internal-token", cfg.ImportAuthToken)
+	res, err := httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+	if e.EventType == "error" && status != failed {
+		err := e.putJobStatus(httpClient, cfg)
+		if err != nil {
+			return err
+		}
+	}
+	defer res.Body.Close()
+
+	err = errorhandler(res.StatusCode)
+	if err != nil {
+		return err
+	}
+	return nil
+
+}
+
+func (e *EventReport) checkInstance(httpClient *http.Client, cfg *config.Config) (string, *[]instanceEvent, error) {
+	path := cfg.ImportAPIURL + "/instances/" + e.InstanceID
+	event := &[]instanceEvent{}
+	var URL *url.URL
+
+	URL, err := url.Parse(path)
+	if err != nil {
+		return "", event, err
+	}
+	res, err := http.Get(URL.String())
+	if err != nil {
+		return "", event, err
+	}
+
+	defer res.Body.Close()
+
+	var instance Instance
+	// json := json.NewDecoder(res.Body).Decode(&Instance{})
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return "", event, err
+	}
+	if err := json.Unmarshal(body, &instance); err != nil {
+		return "", event, err
+	}
+	err = errorhandler(res.StatusCode)
+	if err != nil {
+		return "", event, err
+	}
+	return instance.State, instance.Events, nil
+}
+
+// This will put a error status in the state
+func (e *EventReport) putJobStatus(httpClient *http.Client, cfg *config.Config) error {
+
+	path := cfg.ImportAPIURL + "/instances/" + e.InstanceID
+	log.Info("Instance id: "+e.InstanceID, log.Data{"instance_id": e.InstanceID})
+
+	var URL *url.URL
+
+	URL, err := url.Parse(path)
+	if err != nil {
+		return err
+	}
+	errorhandle := failed
```

Comment:
does this need to be assigned to a separate var, could it not just be `State: failed`?